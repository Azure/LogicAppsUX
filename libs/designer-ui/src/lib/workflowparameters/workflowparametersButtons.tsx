import type { EventHandler } from '../eventhandler';
import type { WorkflowParameterDefinition } from './workflowparameter';
import { Button, Tooltip } from '@fluentui/react-components';
import { bundleIcon, Delete24Filled, Delete24Regular, Edit24Filled, Edit24Regular } from '@fluentui/react-icons';
import { useIntl } from 'react-intl';

const DeleteIcon = bundleIcon(Delete24Filled, Delete24Regular);
const EditIcon = bundleIcon(Edit24Filled, Edit24Regular);

interface ButtonProps {
  onClick: React.MouseEventHandler<HTMLButtonElement>;
}

interface WorkflowParameterDeleteEvent {
  id: string;
}

type WorkflowParameterDeleteHandler = EventHandler<WorkflowParameterDeleteEvent>;

export interface EditOrDeleteButtonProps {
  showDelete?: boolean;
  onDelete?: WorkflowParameterDeleteHandler;
  definition: WorkflowParameterDefinition;
  setIsEditable: (value: boolean | ((prevVar: boolean | undefined) => boolean)) => void;
  setExpanded: (value: boolean | ((prevVar: boolean | undefined) => boolean)) => void;
}

export const EditOrDeleteButton = ({
  showDelete,
  onDelete,
  definition,
  setIsEditable,
  setExpanded,
}: EditOrDeleteButtonProps): JSX.Element => {
  const handleDelete = (e: React.MouseEvent<HTMLButtonElement>): void => {
    if (onDelete) {
      e.stopPropagation();
      onDelete({ id: definition.id });
    }
  };

  const handleEdit = (): void => {
    setIsEditable(true);
    setExpanded(true);
  };
  return showDelete ? <DeleteButton onClick={handleDelete} /> : <EditButton onClick={handleEdit} />;
};

function DeleteButton({ onClick }: ButtonProps): JSX.Element {
  const intl = useIntl();

  const deleteTitle = intl.formatMessage({
    defaultMessage: 'Delete Parameter',
    id: '5rkegy',
    description: 'Delete Button Tooltip Text',
  });

  return (
    <Tooltip relationship="label" content={deleteTitle}>
      <Button
        appearance="subtle"
        aria-label={deleteTitle}
        onClick={onClick}
        icon={<DeleteIcon style={{ color: 'var(--colorBrandForeground1)' }} />}
      />
    </Tooltip>
  );
}

function EditButton({ onClick }: ButtonProps): JSX.Element {
  const intl = useIntl();

  const editTitle = intl.formatMessage({
    defaultMessage: 'Edit parameter',
    id: '03RO5d',
    description: 'Edit Button Tooltip Text',
  });

  return (
    <Tooltip relationship="label" content={editTitle}>
      <Button
        appearance="subtle"
        data-testid="parameter-edit-icon-button"
        data-automation-id="parameter-edit-icon-button"
        aria-label={editTitle}
        onClick={onClick}
        icon={<EditIcon style={{ color: 'var(--colorBrandForeground1)' }} />}
      />
    </Tooltip>
  );
}
