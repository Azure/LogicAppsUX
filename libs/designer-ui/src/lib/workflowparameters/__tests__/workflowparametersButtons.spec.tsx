import type { EditOrDeleteButtonProps } from '../workflowparametersButtons';
import { EditOrDeleteButton } from '../workflowparametersButtons';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import { describe, vi, beforeEach, afterEach, it, expect } from 'vitest';
describe('ui/workflowparameters/workflowparametersButtons', () => {
  let minimal: EditOrDeleteButtonProps;
  let renderer: ReactShallowRenderer.ShallowRenderer;

  beforeEach(() => {
    minimal = {
      //   showDelete: true,
      definition: { id: 'test' },
      setIsEditable: vi.fn(),
      setExpanded: vi.fn(),
    };
    renderer = ReactShallowRenderer.createRenderer();
  });

  afterEach(() => {
    renderer.unmount();
  });

  it('should construct.', () => {
    const button = renderer.render(<EditOrDeleteButton {...minimal} />);
    expect(button).toMatchSnapshot();
  });

  it('should render EditButton when no prop provided.', () => {
    renderer.render(<EditOrDeleteButton {...minimal} />);
    const button = renderer.getRenderOutput();
    const { onClick: clickHandler } = button.props;
    expect(clickHandler).toBeDefined();
  });

  it('should render DeleteButton when no prop provided.', () => {
    const props = { ...minimal, showDelete: true };
    renderer.render(<EditOrDeleteButton {...props} />);
    const button = renderer.getRenderOutput();
    const { onClick: clickHandler } = button.props;
    expect(clickHandler).toBeDefined();
  });
});
