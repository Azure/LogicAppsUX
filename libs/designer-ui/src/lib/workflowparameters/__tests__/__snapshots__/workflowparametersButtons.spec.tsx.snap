// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/workflowparameters/workflowparametersButtons > should construct. 1`] = `
<EditButton
  definition={
    {
      "id": "test",
    }
  }
  onClick={[Function]}
/>
`;
